tenant/module.go

```diff
@@ -2,17 +2,16 @@ package tenant
 
 // Module is the structure of a .Module.yaml file.
 type Module struct {
-	Name       string           `yaml:"name" json:"name"`
-	Namespace  string           `yaml:"namespace" json:"namespace"`
-	Lang       string           `yaml:"lang" json:"lang"`
-	Ref        string           `yaml:"ref" json:"ref"`
-	DraftRef   string           `yaml:"draftRef,omitempty" json:"draftRef,omitempty"`
-	APIVersion string           `yaml:"apiVersion,omitempty" json:"apiVersion,omitempty"`
-	FQMN       string           `yaml:"fqmn,omitempty" json:"fqmn,omitempty"`
-	URI        string           `yaml:"uri" json:"uri,omitempty"`
-	Revisions  []ModuleRevision `yaml:"revisions" json:"revisions"`
-	WasmRef    *WasmModuleRef   `yaml:"-" json:"wasmRef,omitempty"`
-	TokenHash  []byte           `yaml:"-" json:"-"`
+	Name           string           `yaml:"name" json:"name"`
+	Namespace      string           `yaml:"namespace" json:"namespace"`
+	Lang           string           `yaml:"lang" json:"lang"`
+	Ref            string           `yaml:"ref" json:"ref"`
+	DraftRef       string           `yaml:"draftRef,omitempty" json:"draftRef,omitempty"`
+	APIVersion     string           `yaml:"apiVersion,omitempty" json:"apiVersion,omitempty"`         // the version of the API / SDK that this module was built with
+	HostEnvVersion int              `yaml:"hostEnvVersion,omitempty" json:"hostEnvVersion,omitempty"` // the version of the 'host environment' this module needs (0 = before Jan 2023 [legacy], 1 = after Jan 2023)
```

Comment:
I'm not willing to block on this but to me the "HostEnvVersion" can and should be derived from the ApiVersion. 

What if instead of baking this into the module metadata we let the host define what range of API Versions it's capable of handling and thus willing to accept. 

Later we could introduce a first class abstraction for "host runtime" with its own set of metadata which we could use to perform more sophisticated orchestration across a fleet of e2core runtimes.

Reply:
After thinking about this, you're correct and I removed it.